### PR TITLE
Refactor training pipeline

### PR DESCRIPTION
**Pull request by wiederm**
## Description
Reproducing the published performance metrics with the implemented neural networks is difficult. Even for networks that match their original implementation, there is variability in the training routine that seems to effect the training performance significantly. e.g., the reported training performance of SchNet on QM9 is ~0.3 kcal/mol, while the modelforge SchNet currently has an [average test set performance](https://github.com/choderalab/modelforge/wiki/Experiments) of 1.5 kcal/mol.

In this PR I am working through the different approaches that have been reported to improve training performance.

### Normalize dataset

Talking with colleagues and reading this [work](https://arxiv.org/abs/2308.08859) one issue that is often raised is the energy unit dependency of training performance. The hypothesis is that the closer the distribution of $y$ is matched by the initial distribution of the network parameters the easier it is to converge training. Since we opted for kJ/mol, and the actual units don't matter, I will normalize energies for training (dividing by the standard deviation of the energy across the dataset and subtracting the energy mean).

Considerations:
- We opted to directly modify the dataset **before** training as the general approach. Initially, the only operations that were performed were shifting operations (e.g., subtracting atomic self-energies). This had no further impact on calculations besides a gain in efficiency. By normalizing energies units are changed. We can calculate $\hat{E}$  and $L(\hat{E}, E)$ on the normalized energies, but as soon as we mix these with a force loss it is unclear how to best handle this. I think the best approach to avoid any further complications is to calculate $\Delta(\hat{E},E)$ on the normalized energies, but then multiply by the dataset stddev. Also, if we calculate the force loss we perform this calculation on (\hat{E} * E_stddev).

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [ ] TODO 1

## Questions
- [ ] Question1

## Status
- [ ] Ready to go

### CODE REVIEW

**Comment by wiederm** [comment]
It appears as if the difference in training performance reported in the description of this PR can be attributed to the default behavior of the PyTorch dataset, which has been changed in PR #129 .

I will port some of the changes proposed in this PR in a new, separate PR, but I would like to avoid normalization by default if possible. I will leave this PR open in the meantime.

**Comment by wiederm** [comment]
After multiple training repeats, I can confirm that `modelforge` SchNet approaches the reported MAE in the test set performance without normalizing energies.
